File: src/Domain/Entities/Common/GeoCoordinates.cs

namespace BjjWorld.Domain.Entities.Common;

public class GeoCoordinates
{
    public GeoCoordinates()
    {
        Type = "Point";
        Coordinates = [0, 0];
    }

    public GeoCoordinates(double longitude, double latitude)
    {
        Type = "Point";
        Coordinates = [longitude, latitude];
    }

    [BsonElement("type")]
    public string Type { get; private set; } = "Point";

    [BsonElement("coordinates")]
    public List<double> Coordinates { get; set; }
}

File: src/bjjeire-api/src/main/java/com/bjjeire/api/common/GeoCoordinates.java
package com.bjjeire.api.common;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public record GeoCoordinates(String type, List<Double> coordinates, String placeName, String placeId) {
    public GeoCoordinates {
        if (type == null || type.isBlank()) {
            type = "Point";
        }
    }

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    public double latitude() {
        return coordinates != null && coordinates.size() > 1 ? coordinates.get(1) : 0.0;
    }

    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    public double longitude() {
        return coordinates != null && !coordinates.isEmpty() ? coordinates.get(0) : 0.0;
    }
}
